### PR TITLE
[TensorExpr] Use schema instead of op name in NNC lowerings.

### LINKED ISSUE #64963: NNC should rely on schema rather than on node kind in lowerings
## 🐛 Bug

Currently NNC relies on node kind to figure out how to lower a TorchScript node to TE IR:
https://github.com/pytorch/pytorch/blob/3710edc86b6a68a5b77c977f33c8584325accf1f/torch/csrc/jit/tensorexpr/kernel.cpp#L1350-L1351

Instead, we should rely on an op schema because the same node kind can correspond to two variant of an op requiring different lowering (e.g. `min(tensor, tensor)` is a pointwise op while `min(tensor, 1)` is a reduction).

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65555
* #65859
* #65852
* __->__ #65843
* #65554

Fixes #64963.

Differential Revision: [D31282334](https://our.internmc.facebook.com/intern/diff/D31282334)

### CODE REVIEW

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

File: torch/csrc/jit/tensorexpr/lowerings.h [hunk 1]
```diff
@@ -36,9 +36,13 @@ TORCH_API std::unordered_map<std::string, NNCLoweringFunction>&
 getNNCLoweringRegistry();
 TORCH_API NNCLoweringFunction getStandardLoweringFor(const std::string& op);
 
-struct RegisterNNCLoweringFunction {
-  RegisterNNCLoweringFunction(const std::string& name, NNCLoweringFunction fn) {
-    getNNCLoweringRegistry()[name] = fn;
+struct RegisterNNCLoweringsFunction {
+  RegisterNNCLoweringsFunction(
+      const std::vector<std::string>& names,
```
Comment (navahgar) [inline torch/csrc/jit/tensorexpr/lowerings.h:41]: nit: s/names/schemas

**Review by navahgar — APPROVED** [review]
I trust that you have gotten all the schema variants for the ops we have. I didn't go through each one to check that. 

Overall LGTM

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

**Pull request closed without merging**